[WV][91415418] Allow overrides to attributes

warrenv opened the pull request:
[Story](https://www.pivotaltracker.com/story/show/91415418)

This properly handles overrides from the caller by implementing a
deep-copy strategy for merging objects.

Updated to use flightUtils.push and heavily commented.

warrenv committed:
[WV][91415418] Allow overrides to attributes

  This properly handles overrides from the caller by implementing a
deep-copy strategy for merging objects.

mmaiza commented on app/coffeescript/components/mixins/with_default_attributes.coffee:
```diff
@@ -5,19 +5,29 @@ define [
+        else
+          destination[property] = source[property]
+
   return withDefaultAttr
```
We should not clobber. If we need a deep merge let's implement it using flightjs utils.push (better code). If we need to clobber let us have 2 functions.
utils.push(destination, source)

warrenv committed:
[WV][91415418] Addressed PR comments

  - Refactored withDefaultAttributes because the existing implementation did
   not allow overrides.

pasha commented:
:+1:
